### ISSUE TITLE
Changelog not generating when using github squash & merge

### ISSUE THREAD

**Issue by bavedarnow** [commitizen-tools/commitizen#651]
### Description

When merging pull requests using githubs "[squash & merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)" feature, it seems like the cz bump processes correctly, but changelog generation does not.

Doing a `cz check` against the range fails the commit validation. The only solution I have is to disable "squash and merge", as well as "create a merge commit", and only enabling "rebase and merge"

### Steps to reproduce

Create a github pull request with two `feat` commits
Merge the pull request using "Squash and merge"
After merge, the commit looks like this
```
this is a new feature (#1234)

feat: some feature

feat: some other feature
```
Run cz bump and generate a changelog `cz bump --dry-run --changelog`

### Current behavior

The version is bumped, but the changelog does not generate for the messages in the squash

```
bump: version 0.0.1 → 0.1.0
tag to create: 0.1.0
increment detected: MINOR

## 0.1.0 (2023-01-06)
```

### Desired behavior

```
bump: version 0.0.1 → 0.1.0
tag to create: 0.1.0
increment detected: MINOR

## 0.1.0 (2023-01-06)

### Feat

- some feature
- some other feature
```

### Environment

Commitizen Version: 2.39.1
Python Version: 3.7.16 (default, Jan  3 2023, 10:02:01)
[Clang 14.0.0 (clang-1400.0.29.202)]
Operating System: Darwin

**Comment by Lee-W** [comment]
I don't think commitizen is designed to handled squashed commit. (@woile please let me know if I'm wrong.) but it's a bit weird it cannot work with "create a merge commit". could you please provide an example on this one so that we can have a better idea? Thanks!

**Comment by woile** [comment]
Using squash works. Look at [this commit](https://github.com/commitizen-tools/commitizen-action/commit/35d0a042570abea72fb50570bdaf9269e1c45d78).

The problem is that the "squashed" commit, does not follow conventional commits. Your example:

```
this is a new feature (#1234)

feat: some feature

feat: some other feature
```

is not a valid conventional commit.
When squashing,  you should change the summary of the commit. Maybe do something like:

```
feat: some feature

feat: some other feature

Closes #1234
```
